### PR TITLE
[RF-1409] Add customer types

### PR DESCRIPTION
**Pull request by alyant34**
## Ticket
- https://rizemoney.atlassian.net/browse/RF-1409

## Changes
- Add customer type and business name to the Customer object

### CODE REVIEW

**Comment by alyant34** [comment]
@ksalmon I am sure I will need to make additional changes, but I am not sure why all the checks were not run?

File: types/lib/core/customer.d.ts [hunk 1]
```diff
@@ -26,16 +26,17 @@ declare class CustomerService {
...
+    protected _validateCreateParams(externalUid: string, email: string, customer_type: 'unaffiliated' | 'sole_proprietor'): void;
     /**
      * @ignore @protected
      * Validates the parameters for the "update" method
      * @param {string} uid
      * @param {string} email
      * @param {CustomerDetails} details
      */
-    protected _validateUpdateParams(uid: string, email: string, details: CustomerDetails): void;
+    protected _validateUpdateParams(uid: string, email: string, details: CustomerDetails, customer_type?: string): void;
```
Comment (ksalmon) [inline types/lib/core/customer.d.ts:39]:
```suggestion
    protected _validateUpdateParams(uid: string, email: string, details: CustomerDetails, customer_type?: 'unaffiliated' | 'sole_proprietor'): void;
```

Comment (ksalmon) [reply types/lib/core/customer.d.ts:39]: I would create a type for this like `type CustomerType = 'unaffiliated' | 'sole_proprietor'`

**Review by ksalmon — CHANGES_REQUESTED** [review]

File: test/core/custodial-account.spec.js [hunk 2]
```diff
@@ -21,56 +21,71 @@ describe('Custodial Accounts', () => {
...
         it('Throws an error if "liability" query parameter is invalid', () => {
-            const promise = rizeClient.custodialAccount.getList({liability: ' '});
-            return expect(promise).to.eventually.be.rejectedWith('"liability" query must be a boolean.');
+            const promise = rizeClient.custodialAccount.getList({ liability: ' ' });
+            return expect(promise).to.eventually.be.rejectedWith(
+                '"liability" query must be a boolean.'
+            );
         });
 
         it('Throws an error if "type" query parameter is invalid', () => {
```
[the remark this answers is not in the document]
Comment (alyant34) [reply test/core/custodial-account.spec.js:75]: I am not sure I know what you mean, I tested the customer_type query in the customer.spec.js file

**Pull request merged**